File: leetcode/solutions/problem0001/solution.py
# Solution for problem 1
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_map = {}
        for idx in range(len(nums)):
            target_num = target-nums[idx]
            if target_num in num_map:
                return [num_map[target_num], idx]
            num_map[nums[idx]] = idx

File: leetcode/solutions/problem0001/solution.java
import java.util.HashMap;
import java.util.Map;

public class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> num_map = new HashMap<>();
        for (int idx = 0; idx < nums.length; idx++) {
            int target_num = target - nums[idx];
            if (num_map.containsKey(target_num)) {
                return new int[] { num_map.get(target_num), idx };
            }
            num_map.put(nums[idx], idx);
        }
        throw new IllegalArgumentException("No solution found");
    }
}
